x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/arithmetic/Sub.java

```diff
@@ -34,7 +34,7 @@ protected Sub replaceChildren(Expression newLeft, Expression newRight) {
 
     @Override
     protected TypeResolution resolveWithIntervals() {
-        if (right().dataType().isDateBased() && DataTypes.isInterval(left().dataType())) {
+        if ((right().dataType().isDateBased() || right().dataType().isTimeBased()) && DataTypes.isInterval(left().dataType())) {
```

Comment:
There are many calls for `dataType().isDateBased() || dataType().isTimeBased()`. I think it's worth adding to DataType a method that should do this check. Maybe called `isTemporalBased`?